featuretools/synthesis/encode_features.py

```diff
@@ -113,7 +118,12 @@ def encode_features(feature_matrix, features, top_n=10, include_unknown=True,
         val_counts = val_counts.sort_values([f.get_name(), index_name],
                                             ascending=False)
         val_counts.set_index(index_name, inplace=True)
-        unique = val_counts.head(top_n).index.tolist()
+        select_n = top_n
+        if drop_first:
+            if len(val_counts) != 1 or top_n != 1:
```

Comment:
what if we remove the conditional and just do these two lines?

```
select_n = min(len(val_counts), top_n) 
select_n = max(select_n - 1, 1)  # make sure at least 1 category is selected
```

Reply:
@kmax12 
But we have to keep outer conditional statement ( if drop_first ).

Reply:
yep, sorry, I was just referring to inner conditional